src/main/java/uk/gov/ida/matchingserviceadapter/mappers/MatchingDatasetToMatchingDatasetDtoMapper.java

```diff
@@ -114,4 +115,9 @@ public VerifyAddressDto apply(Address input) {
                         input.isVerified()))
                 .collect(Collectors.toList());
     }
+
+    public static Comparator<TransliterableMdsValue> getFirstNameComparator() {
+        return Comparator.comparing(TransliterableMdsValue::isVerified, Comparator.reverseOrder())
+                .thenComparing(TransliterableMdsValue::getTo, Comparator.nullsFirst(null));
+    }
```

Comment:
Wasn't there talk of including the from date in the sort?

Reply:
So Chris added that to the [VSP](https://github.com/alphagov/verify-service-provider/blob/master/src/main/java/uk/gov/ida/verifyserviceprovider/mappers/MatchingDatasetToNonMatchingAttributesMapper.java#L145-L147), but I'm not sure it provides any benefit here does it? Yes it would make the ordering more complete. But the bug fix was to use a verified, current name, and the from date doesn't affect that.

Reply:
I agree with Alex - we only really care about the `to` date - and there should only ever be one current verified name (i.e. `verified == true && toDate == null`). (We're not checking that, though....)